src/solidity-frontend/solidity_grammar.cpp

```diff
@@ -65,21 +66,28 @@ TypeNameT get_type_name_t(const nlohmann::json &type_name)
   if(type_name.contains("typeString"))
   {
     // for AST node that contains ["typeName"]["typeDescriptions"]
-    if(type_name["typeString"] == "uint8" || type_name["typeString"] == "bool")
+    std::set<std::string> uintTypeStringSet = {
```

Comment:
make it `static` in namespace SolidityGrammar?

Reply:
> TBH, I don't understand this PR:
> 
> 1. `convert_integer_literal` takes an `const nlohmann::json &integer_literal` but somehow we are passing a `literal_type`?
> 2. Why do we need the type after all? Since you're passing the expr around, you should be able to retrieve it whenever you want.

1. This is a result of the existing structure, `convert_integer_literal` seems to also handle the creating of of the variables of uint types. The way the AST is built is that in an initialization with a literal rvalue `uint8 i = 10` we get type definitions for the left side but for the right side we get a generic integer type without any indication for what subtype of the integer it is (unsigned or signed, bitsize) which then requires us to pass in the type from the left hand side which is what the literal_type stores here
2. This is true that we are passing the expression about but because literal expressions do not contain any specific type descriptions we need to pass them around from the lhs of the expression.